pjsip/include/pjsua2/media.hpp

```diff
@@ -426,6 +426,75 @@ typedef std::vector<AudioMedia*> AudioMediaVector;
 /** Array of Audio Media */
 typedef std::vector<AudioMedia> AudioMediaVector2;
 
+/**
+ * This structure describes a media frame.
+ */
+struct MediaFrame
+{
+    pjmedia_frame_type   type;      /**< Frame type.                        */
+    string               buf;       /**< Frame buffer content.              */
+    unsigned             size;      /**< Frame size in bytes.               */
+
+public:
+    /**
+     * Default constructor
+     */
+    MediaFrame()
+    : type(PJMEDIA_FRAME_TYPE_NONE),
+      size(0)
+    {}
+};
+
+/**
+ * Audio Media Port.
+ */
+class AudioMediaPort : public AudioMedia
+{
+public:
+    /**
+     * Constructor.
+     */
+    AudioMediaPort();
+
+    /**
+     * Destructor. This will unregister the audio media port from the
+     * conference bridge.
+     */
+    virtual ~AudioMediaPort();
+
+    /**
+     * Create an audio media port and register it to the conference bridge.
+     *
+     * @param name      The port name.
+     * @param fmt       The audio format.
+     */
+    void createPort(const string &name, MediaFormatAudio &fmt)
```

Comment:
Why is this not part of the constructor?

Reply:
I want to make it inline with `AudioMediaPlayer/Recorder` and `ToneGenerator` which all have their own `create...()`